qiskit/algorithms/gradients/lin_comb_estimator_gradient.py

```diff
@@ -73,7 +87,7 @@ def __init__(
                 default options > primitive's default setting.
                 Higher priority setting overrides lower priority setting.
         """
-        self._gradient_circuits = {}
+        self._lin_comb_cache = {}
```

Comment:
Why do we need the `gradient_circuits` cache and `lin_comb_cache`? Doesn't the base class already cache the gradient circuits🤔